src/imageResource.js

```diff
@@ -0,0 +1,132 @@
+  this.needed = config.needed || false;
+  this.visible = config.visible || false;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.x = config.x || 0;
+  this.y = config.y || 0;
+  this.height = config.height || 1;
+  this.width = config.width || 1;
+  this.zIndex = config.zIndex || 0;
+  this.tileSource = config.tileSource;
+  this.dynamic = config.dynamic || false;
+  this.imageType = config.imageType || "main"; // can be 'main', 'alternate', or 'detail'
+
+  this.status = 'initialized'; // can be 'requested', 'received', 'pending','shown', or 'failed'
+  this.dispatcher = dispatcher;
+};
+
+ImageResource.prototype = {
+  hide: function() {
+
+  },
+
+  setOpacity: function(opacity) {
+    this.opacity = opacity;
+    this.tiledImage.setOpacity(opacity);
+  },
+
+  getOpacity: function() {
+    return this.opacity;
+  },
+
+  openTileSource: function(viewer, parentBounds, parentHandler) {
+    var self = this;
+
+    // We've already loaded this tilesource
+    if(this.status === 'shown') {
+      return;
+    }
+
+    // otherwise, continue loading the tileSource.
+    this.dispatcher.emit('image-resource-tile-source-requested', { 'detail': this.tileSource });
+    self.status = 'requested';
+    var position = this._getPositionInViewer(parentBounds);
+    viewer.addTiledImage({
+      x: position.x,
+      y: position.y,
+      width: parentBounds.width * this.width,
+      tileSource: this.tileSource,
+      opacity: this.opacity,
+      clip: this.clipRegion,
+      index: this.zIndex,
+
+      success: function(event) {
+        var main = event.item;
+        self.status = 'pending';
+
+        var tileDrawnHandler = function(event) {
+          if (event.tiledImage === main) {
+            self.tiledImage = main;
+            self.setPosition(parentBounds);
+            self.visible = true;
+            self.status = 'shown';
+            viewer.removeHandler('tile-drawn', tileDrawnHandler);
+            self.dispatcher.emit('image-resource-tile-source-opened', { 'detail': self.tileSource });
+            parentHandler(event);
+          }
+        };
+        viewer.addHandler('tile-drawn', tileDrawnHandler);
+      },
+
+      error: function(event) {
+        var errorInfo = {
+          id: self.tileSource,
+          message: event.message,
+          source: event.source
+        };
+        self.status = 'failed';
+        self.dispatcher.emit('image-resource-tile-source-failed', {'detail': errorInfo});
+      }
+    });
+  },
+
+  getImageType: function() {
+    return this.imageType;
+  },
+
+  getBounds: function() {
+    return new OpenSeadragon.Rect(this.bounds.x, this.bounds.y, this.bounds.width, this.bounds.height);
+  },
+
+  _getPositionInViewer: function(parentBounds) {
+    return new OpenSeadragon.Point(
+        parentBounds.x + (parentBounds.width * this.x),
+        parentBounds.y + (parentBounds.width * this.y));
+  },
+
+  setPosition: function(parentBounds, immediately) {
```

Comment:
I think setPosition is a confusing name for this function...maybe it should be setParentBounds. Of course if we go with just already knowing our parent, then it becomes updateForParentBounds or something. If we want our opacity to be based in part on the parent opacity, maybe we add that in and call it updateForParentChange.